src/App.js

```diff
@@ -0,0 +1,169 @@
+    let item = {
+        id : todoItemID++,
+        status : "view",
+        contents : contents
+    }
+    
+    todoItems.push(item);
+    render(todoItems);
+}
+
+const onChangeTodoInput = (e) => {    
+    if (e.keyCode === 13) {
+        addTodoItem(e.target.value);
+        e.target.value = "";
+    }
+}
+
+const render = (items) => {
+    let template = items.map(item => todoItemTemplate(item));
+    todoList.innerHTML = template.join("");
+
+    todoList.addEventListener('click', onToggleTodoItem);
+    todoList.addEventListener('dblclick', onModifyTodoItem);
+    todoList.addEventListener('keydown', onChangeTodoItem)
+    todoList.addEventListener('click', onDestroyTodoItem);
+    
+    todoCount.innerHTML = todoItems.length;
+
+    localStorage.setItem('todoItems', JSON.stringify(todoItems));
+}
+
+const showAllTodoList = () => {
+    btn_all.classList.add("selected");
+    btn_active.classList.remove("selected");
+    btn_completed.classList.remove("selected");
+
+    render(todoItems);
+}
+
+const showActiveTodoList = () => {
+    btn_all.classList.remove("selected");
+    btn_active.classList.add("selected");
+    btn_completed.classList.remove("selected");
+
+    render(todoItems.filter(item => item.status !== "completed"));
+}
+
+const showCompletedTodoList = () => {
+    btn_all.classList.remove("selected");
+    btn_active.classList.remove("selected");
+    btn_completed.classList.add("selected");
+
+    render(todoItems.filter(item => item.status === "completed"));
+}
+
+const onToggleTodoItem = (e) => {
+    if (e.target.className === "toggle") {
+        todoItems = todoItems.map(
+            item => {
+                if (item.id == e.target.id) {
+                    return item = {
+                        ...item,
+                        status : "completed"
+                    }
+                } else {
+                    return item;
+                }
+            }
+        );
+
+        render(todoItems);
+    }
+}
+
+const onDestroyTodoItem = (e) => {
+    if (e.target.className === "destroy") {
+        if (confirm("정말 삭제하시겠습니까?") == true) {            
+            todoItems = todoItems.filter(item => item.id != e.target.id);
+            
+            render(todoItems);
+        } else {
+            return;
+        }
+    }
+}
+
+const onModifyTodoItem = (e) => {
+    if (e.target.className === "label") {
+        todoItems = todoItems.map(
+            item => {
+                if (item.id == e.target.id) {
+                    return item = {
+                        ...item,
+                        beforeStatus : item.status,
+                        status : "editing"
+                    }
+                } else {
+                    return item;
+                }
+            }
+        );
+
+        render(todoItems);
+    }
+}
+
+const onChangeTodoItem = (e) => {
+    if (e.target.className === "edit") {
+        if (e.keyCode === 13) {
+            if (e.target.value !== "") {
```

Comment:
제가 위에서 언급한 방법처럼 해당하는 케이스가 아닌 경우 return을 통해 바로 함수를 빠져나가도록 하거나 조건을 하나로 합치면 좋을 것 같아요.
```suggestion
    if (e.target.className === "edit" && e.keyCode === 13 && e.target.value !== "") {
```
3중첩이 한개로 처리 가능하네요! if 조건절을 함수로 분리해서 `isChangableEvent` 이런식으로 의미를 부여할 수도 있겠네요.

Reply:
감사해요! if 중첩을 최대한 줄여보는 습관을 가져보겠습니다 👼